### PR TITLE
(feat) - O3-3197 - initial ward app

### PR DESCRIPTION
**Pull request by chibongho**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

Not much here. Just initial files for a new ward app.

### CODE REVIEW

**Review by mogoodrich — APPROVED** [review]
LGTM, though generally defer to others...

File: packages/esm-ward-app/src/createDashboardLink.component.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React, { useMemo } from 'react';
```
Comment (mseaton) [inline packages/esm-ward-app/src/createDashboardLink.component.tsx:1]: Remove this component.  If we need to add this to the home dashboard component in the future, and things are still not refactored out of the home app, we can add something like this in.  I'm hoping it will not be needed.

File: packages/esm-ward-app/src/dashboard.meta.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export const wardViewDashboardMeta = {
```
Comment (mseaton) [inline packages/esm-ward-app/src/dashboard.meta.ts:1]: Remove this file

File: packages/esm-ward-app/src/home.comonent.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
```
Comment (mseaton) [inline packages/esm-ward-app/src/home.comonent.tsx:1]: This filename is spelled incorrectly.  Additionally, I'd rather not create components that have no basis in reality yet.  We have no design for a "Home" page in the ward app.  Get rid of this component

File: packages/esm-ward-app/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+const moduleName = '@openmrs/esm-ward-app';
+
+const options = {
+  featureName: 'ward',
+  moduleName,
+};
+
+export const root = getSyncLifecycle(rootComponent, options);
+
+// TODO: uncomment this for the link to the ward app to appear on the left hand nav when the app is more ready
```
Comment (mseaton) [inline packages/esm-ward-app/src/index.ts:17]: Remove this commented code

File: packages/esm-ward-app/src/root.component.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import Home from './home.comonent';
+
+const Root: React.FC = () => {
+  const wardViewBasename = window.getOpenmrsSpaBase() + 'ward';
+
+  return (
+    <main>
+      <BrowserRouter basename={wardViewBasename}>
+        <Routes>
+          <Route path="/" element={<Home />} />
```
Comment (mseaton) [inline packages/esm-ward-app/src/root.component.tsx:12]: Per above, there is no current use case to have a Home component at the root of this app.  Our initial view that we are designing out will be a component that renders the patients who are currently assigned to a particular ward.  So let's create that.  Let's put that at route `/ward/${locationUuid}/patients/`, and renders a component defined as `<WardPatients location={locationUuid} />`.  Please create this WardPatients component.  You can render the location name in the top left corner as a starting point, per the mockups.

**Review by mseaton — CHANGES_REQUESTED** [review]

**Review by mseaton — COMMENTED** [review]
Please ensure that the initial ward app we are committing has the following:

* Framework setup for unit tests, with an initial unit test for the initial page rendering
* Config schema (empty to start}

File: packages/esm-ward-app/src/routes.json [hunk 6]
```diff
@@ -0,0 +1,29 @@
+{
+  "$schema": "https://json.openmrs.org/routes.schema.json",
+  "backendDependencies": {
+    "webservices.rest": "^2.2.0"
+  },
+  "extensions": [
+    {
+      "component": "root",
```
Comment (ibacher) [inline packages/esm-ward-app/src/routes.json:8]: It's definitely unorthodox to have the same component as both an extension and a page.

File: yarn.lock.orig
Comment (ibacher) [inline yarn.lock.orig:1]: This file should almost certain not be part of this PR...

**Comment by chibongho** [comment]
Updates:
- added tests and config for the ward app.
- make the page render the location name, taken from either the location uuid in URL (if provided) or the session location.
- removed accidental added yarn.lock.orig file, stuff related to dashboard link.

image
image

**Review by mseaton — APPROVED** [review]
Looks great, thanks.  I think in the ward-view, you could have used the opportunity where you have the <h1> to introduce a ward-view css file, but we can introduce that as we go.

File: packages/esm-ward-app/package.json [hunk 7]
```diff
@@ -0,0 +1,53 @@
+{
+  "name": "@openmrs/esm-ward-app",
+  "version": "7.0.0",
+  "description": "Ward and bed management module for the OpenMRS SPA",
```
Comment (denniskigen) [inline packages/esm-ward-app/package.json:4]:
```suggestion
  "description": "Ward and bed management module for O3",
```

**Review by denniskigen — APPROVED** [review]
Great start! Thanks
